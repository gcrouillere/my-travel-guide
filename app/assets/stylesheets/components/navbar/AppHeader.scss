.navbar {
  box-shadow: 0 4px 6px rgba(50,50,93,0.11), 0 1px 3px rgba(0,0,0,0.08);
  z-index: 5;
  .menu {
    display: flex;
    justify-content: center;
    p {
      margin: 0;
      cursor: pointer;
    }
    .nav-link {
      text-align: center;
      display: flex;
      justify-content: center;
      align-items: center;
       @media all and (max-width:768px) {
        font-size: 12px;
       }
    }
  }
}
