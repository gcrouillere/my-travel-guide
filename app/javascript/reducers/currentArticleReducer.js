import { FETCH_ARTICLE } from '../actions'

export default function(state = null, action) {

  switch(action.type) {
    case FETCH_ARTICLE:
      return action.payload
    default:
      return state
  }
}
