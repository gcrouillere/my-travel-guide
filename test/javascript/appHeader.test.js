import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import { BrowserRouter as Router, Route } from 'react-router-dom'
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import { createStore, combineReducers, applyMiddleware } from 'redux';

import AppHeader from '../../app/javascript/packs/appHeader'

configure({ adapter: new Adapter() });

import currentUserReducer from '../../app/javascript/reducers/currentUserReducer'
import currentArticleReducer from '../../app/javascript/reducers/currentArticleReducer'

const reducers = combineReducers({
  currentUser: currentUserReducer,
  currentArticle: currentArticleReducer
})

describe('AppHeader test Suite', () => {
  it('renders component', () => {
    const appHeaderTree = renderer.create(
      <Provider store={createStore(reducers)}>
        <Router>
          <AppHeader location={{pathname: "/articles"}}/>
        </Router>
      </Provider>
    ).toJSON()
    expect(appHeaderTree).toMatchSnapshot()
  })
})
