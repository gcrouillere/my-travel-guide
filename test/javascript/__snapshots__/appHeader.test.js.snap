// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader test Suite renders component 1`] = `
<div>
  <nav
    className="navbar navbar-light bg-dark"
  >
    <div
      className="brand"
    >
      <a
        className="navbar-brand nav-link text-white"
        href="/"
        onClick={[Function]}
      >
        My Travel Guide
      </a>
    </div>
    <div
      className="menu"
    >
      <a
        className="nav-link text-white"
        href="/articles/new"
        onClick={[Function]}
      >
        Create an article
      </a>
    </div>
  </nav>
</div>
`;
